internal/impl/elasticsearchv8/output.go

```diff
@@ -0,0 +1,373 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package elasticsearchv8
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/elastic/elastic-transport-go/v8/elastictransport"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/typedapi/core/bulk"
+	"github.com/elastic/go-elasticsearch/v8/typedapi/types"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	esFieldURLs            = "urls"
+	esFieldID              = "id"
+	esFieldAction          = "action"
+	esFieldIndex           = "index"
+	esFieldPipeline        = "pipeline"
+	esFieldRouting         = "routing"
+	esFieldRetryOnConflict = "retry_on_conflict"
+	esFieldTLS             = "tls"
+	esFieldAuth            = "basic_auth"
+	esFieldAuthEnabled     = "enabled"
+	esFieldAuthUsername    = "username"
+	esFieldAuthPassword    = "password"
+	esFieldBatching        = "batching"
+)
+
+type esConfig struct {
+	clientOpts elasticsearch.Config
+
+	actionStr       *service.InterpolatedString
+	idStr           *service.InterpolatedString
+	indexStr        *service.InterpolatedString
+	pipelineStr     *service.InterpolatedString
+	routingStr      *service.InterpolatedString
+	retryOnConflict int
+}
+
+func esConfigFromParsed(pConf *service.ParsedConfig) (*esConfig, error) {
+	conf := &esConfig{}
+
+	if os.Getenv("ELASTICSEARCH_DEBUG") != "" {
+		conf.clientOpts.Logger = &elastictransport.CurlLogger{
+			Output:             os.Stdout,
+			EnableRequestBody:  true,
+			EnableResponseBody: true,
+		}
+	}
```

Comment:
Can we wrap this logger with our service logger at trace level instead?

Reply:
There's not an easy way to do that, `elastictransport.CurlLogger` is a decorator around a custom roundtripper type interface. I could write my own struct for that but it doesn't really seem worth it for a debugger that I intended to really only be used by us during development.